src/ok.rs

```diff
@@ -114,6 +135,23 @@ const SEV_ES_CPUIDS: &[CpuId; 1] = &[CpuId {
 pub fn cmd(gen: Option<SevGeneration>, quiet: bool) -> Result<()> {
     let mut passed = true;
 
+    let cpuid_vec = match gen {
+        Some(g) => collect_cpuids(g),
+        None => collect_cpuids(SevGeneration::Es),
+    };
+
+    for list in cpuid_vec {
+        for r in list {
+            let res = r.execute();
+            if !check_test(&res, quiet) {
+                passed = false;
+            }
+        }
+        if !passed {
+            break;
+        }
```

Comment:
Why break here?

Reply:
The `break` is used to ensure that useless information isn't printed to the console. Each `list` in `cpuid_vec` is a higher step in the CPUID hierarchy. For example, if the processor is found to not be from AMD, then `passed` would be changed to `false` while traversing `CORE_CPUID`. Subsequently, we don't want to test anymore CPUIDs, as the results are useless.